src/transaction/log_page_buffer.c

```diff
@@ -6092,6 +6096,33 @@ logpb_remove_archive_logs_exceed_limit (THREAD_ENTRY * thread_p, int max_count)
 		  assert (false);
 		}
 	    }
+
+	  /* check flashback */
+	  if (flashback_is_needed_to_keep_archive ())
+	    {
+	      flashback_first_pageid = flashback_min_log_pageid_to_keep ();
```

Comment:
There seems to be a race condition. After `flashback_is_needed_to_keep_archive`()` return true, `flashback_Min_log_pageid` can be set `NULL_LOG_PAGEID` because a flashback request faces an error or finishes. What do you think?

Reply:
yes, you are right. 
I need to consider that case, and add condition in line 6108 
`if (flashback_first_pageid != NULL_LOG_PAGEID && logpb_is_page_in_archive (flashback_first_pageid)) `

Reply:
How about just using mutex and try_lock instead of atomic variable and the active variable. It seems that this is not performance-intensive code, which means we don't need to make it lock-free and worry about tricky race conditions.

Reply:
> How about just using mutex and try_lock instead of atomic variable and the active variable. It seems that this is not performance-intensive code, which means we don't need to make it lock-free and worry about tricky race conditions and memory barriers.

I don't think we have enough time to consider another detailed design. Just let it go.

Reply:
I'll put it on my to-do list and put it together when I refactor. thank you.